src/luarocks/fetch.lua

```diff
@@ -355,7 +355,32 @@ function fetch.get_sources(rockspec, extract, dest_dir)
       if not ok then return nil, err end
       fs.unpack_archive(rockspec.source.file)
       if not fs.exists(rockspec.source.dir) then
-         return nil, "Directory "..rockspec.source.dir.." not found inside archive "..rockspec.source.file, "source.dir", source_file, store_dir
+
+         -- if rockspec.source.dir can't be found, see if we only have one
+         -- directory in store_dir.  If that's the case, assume it's what
+         -- we're looking for.
+         -- Ideally, we'd only do this if rockspec.source.dir was not
+         -- defined in the rockspec, but load_local_repository already
```

Comment:
`load_local_rockspec`, you mean.

Wouldn't it be better to set a boolean there (such as `rockspec.source.dir_set`) so we can know here if the value was inferred or explicit?